### PR TITLE
Modified the index with a comment to rebuild the domain index

### PR DESCRIPTION
**Pull request by mjriley**
## Technical Summary
Attempting to rebuild the domains couch index to fix a domain not appearing as actually deleted. We had issues deleting a domain from https://dimagi-dev.atlassian.net/browse/SAAS-14729.

### Safety story
This will rebuild the domains index. A new migration has been created that should build this next index asynchronously, and thus not generate any downtime

### Automated test coverage

No tests

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/preindex/migrations/0002_initial.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.20 on 2023-07-12 20:09
+
+import corehq.preindex.django_migrations
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    initial = True
```
Comment (dannyroberts) [inline corehq/preindex/migrations/0002_initial.py:9]: @millerdev is this supposed to have `initial = True` for every time you run `./manage.py makemigrations preindex` as per https://github.com/dimagi/commcare-hq/blob/master/CONTRIBUTING.rst#reindex--migration?

**Review by dannyroberts — COMMENTED** [review]
This looks good but I had one doubt looking at the migration that was automatically generated

File: corehq/preindex/migrations/0002_initial.py (hunk 1 above)
Comment (millerdev) [reply corehq/preindex/migrations/0002_initial.py:9]: Probably not. I'm not sure why it's doing that.

Comment (mjriley) [reply corehq/preindex/migrations/0002_initial.py:9]: Is this line something I should remove from this PR? Perhaps opening a separate PR to address this afterwards

Comment (millerdev) [reply corehq/preindex/migrations/0002_initial.py:9]: Typically each new migration depends on the previous one, and it's usually automatically generated
```suggestion
    dependencies = [
        ('preindex', '0001_initial'),
    ]
```
It's not clear why that didn't happen here. There's probably no harm in fixing it.

**Review by dannyroberts — APPROVED** [review]

**Pull request merged**